core/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -253,10 +251,9 @@ public  T loadLatestState(ESLogger logger, Path... dataLocations) throws IOExcep
         if (dataLocations != null) { // select all eligable files first
             for (Path dataLocation : dataLocations) {
                 final Path stateDir = dataLocation.resolve(STATE_DIR_NAME);
-                if (!Files.isDirectory(stateDir)) {
-                    continue;
-                }
                 // now, iterate over the current versions, and find latest one
+                // we don't check if the stateDir is present since it could be deleted
+                // after the check. Also if there is a _state file and it's not a dir something is really wrong
                 try (DirectoryStream<Path> paths = Files.newDirectoryStream(stateDir)) { // we don't pass a glob since we need the group part for parsing
```

Comment:
yeah so this is a change over what used to be here.... I think we should really barf since this is really a violation of what we expect - it removes some leniency